Modules/mmapmodule.c

```diff
@@ -273,30 +421,31 @@ mmap_read_line_method(mmap_object *self,
 {
     Py_ssize_t remaining;
     char *start, *eol;
-    PyObject *result;
 
     CHECK_VALID(NULL);
 
     remaining = (self->pos < self->size) ? self->size - self->pos : 0;
     if (!remaining)
         return PyBytes_FromString("");
     start = self->data + self->pos;
-    eol = memchr(start, '\n', remaining);
+
+    if (safe_memchr(&eol, start, '\n', remaining) < 0) {
```

Comment:
```suggestion
    if (safe_memchr((void **)&eol, start, '\n', remaining) < 0) {
```

For the compiler warnings (I hope... I didn't test whether this will actually fix them)

Reply:
I changed the signature of `safe_memchr` to use `char **` instead. Otherwise I am dereferencing a void pointer in the function.